packages/cubejs-api-gateway/src/gateway.ts

```diff
@@ -348,6 +348,23 @@ class ApiGateway {
       })
     );
 
+    app.post(
+      `${this.basePath}/v1/cubesql`,
+      userMiddlewares,
+      userAsyncHandler(async (req, res) => {
+        const server = this.initSQLServer();
+        
+        res.setHeader('Content-Type', 'application/json');
+        res.setHeader('Transfer-Encoding', 'chunked');
+
+        try {
+          await server.execSql(req.body.query, res, req.context?.securityContext);
+        } catch (error: any) {
+          res.status(500).end(JSON.stringify(error));
```

Comment:
It's better to use `this.handleError` because some types of errors can confidential details